test_binheap.py

```diff
@@ -0,0 +1,98 @@
+import pytest
+
+from binheap import BinaryHeap
+
```

Comment:
```python
def test_push_1():
    binheap = BinaryHeap("max")
    binheap.push(4)
    assert binheap._storage == [4]
    binheap.push(3)
    assert binheap._storage == [4, 3]
    binheap.push(5)
    assert binheap._storage == [5, 3, 4]
    binheap.push(8)
    #      5
    #     / \
    #    3   4
    #   /
    #  8
    # heap-down
    # 3 <--> 8, 5 <--> 8
    #      8
    #     / \
    #    5   4
    #   /
    #  3
    assert binheap._storage == [8, 5, 4, 3]
    binheap.push(11)
    #      8
    #     / \
    #    5   4
    #   / \
    #  3   11
    # heap-down
    # 11 <--> 5, 11 <--> 8
    #      11
    #     / \
    #    8   4
    #   / \
    #  3   5
    assert binheap._storage == [11, 8, 4, 3, 5]
    binheap.push(15)
    #      11
    #     /  \
    #    8    4
    #   / \  /
    #  3  5  15
    # heap-down
    # 15 <--> 4, 15 <--> 11
    #      15
    #     /  \
    #    8    11
    #   / \  /
    #  3  5  4
    assert binheap._storage == [15, 8, 11, 3, 5, 4]


def test_push_2():
    binheap = BinaryHeap("max")
    binheap.push(11)
    assert binheap._storage == [11]
    binheap.push(3)
    assert binheap._storage == [11, 3]
    binheap.push(8)
    assert binheap._storage == [11, 3, 8]
    binheap.push(5)
    #      11
    #     / \
    #    3   8
    #   /
    #  5
    # heap-down
    # 5 <--> 3
    #      11
    #     / \
    #    5   8
    #   /
    #  3
    assert binheap._storage == [11, 5, 8, 3]
    binheap.push(4)
    #      11
    #     / \
    #    5   8
    #   / \
    #  3   4
    # heap-down
    # no-op
    assert binheap._storage == [11, 5, 8, 3, 4]
    binheap.push(15)
    #      11
    #     /  \
    #    5    8
    #   / \   /
    #  3  4  15
    # heap-down
    # 15 <--> 8, 15 <--> 11
    #       15
    #     /   \
    #    5    11
    #   / \   /
    #  3  4  8
    assert binheap._storage == [15, 5, 11, 3, 4, 8]
```

Reply:
i think i see what was missing so i added a large test for the min as well